app/Console/Commands/DeviceList.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Console\LnmsCommand;
+use App\Models\Device;
+use Symfony\Component\Console\Input\InputOption;
+
+class DeviceList extends LnmsCommand
+{
+    protected $name = 'device:list';
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+        $this->addOption('apps', 'a', InputOption::VALUE_NONE);
+        $this->addOption('ports', 'p', InputOption::VALUE_NONE);
+        $this->addOption('ip', 'i', InputOption::VALUE_NONE);
+        $this->addOption('storage', 's', InputOption::VALUE_NONE);
+        $this->addOption('sensors', 'S', InputOption::VALUE_NONE);
+        $this->addOption('device-per-line', 'l', InputOption::VALUE_NONE);
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     */
+    public function handle(): int
+    {
+        $devices = Device::when($this->option('apps'), fn ($q) => $q->with('applications'))
+            ->when($this->option('ports'), fn ($q) => $q->with(['ports', 'ports.ipv4', 'ports.ipv6']))
+            ->when($this->option('storage'), fn ($q) => $q->with('storage'))
+            ->when($this->option('storage'), fn ($q) => $q->with('sensors'))
+            ->get();
+
+        if ($this->option('device-per-line')) {
```

Comment:
This is a weird option, why do we want that?

Reply:
For when processing it as a CSV. It simplifies handling if you already know the position of the data you are processing is in it means you don't need to include code to also exclude the header line.

Reply:
wait!

Sorry, derp. Was thinking the headerless bit.

This is if handy if you want to process the output similar to like processing the output in a similar manner to a JSON log file, which is basically a entry per line.